src/main/kotlin/lotto/domain/LottoResult.kt

```diff
@@ -0,0 +1,31 @@
+package lotto.domain
+
+class LottoResult {
+    private var lottoResults = mutableMapOf<LottoRanking, Int>()
+
+    init {
+        lottoResults = LottoRanking.values().associateWith { 0 } as MutableMap<LottoRanking, Int>
+    }
+
+    fun lottoRanking(purchaseLottoNumbers: List<LottoNumbers>, winningLotto: WinningLotto): MutableMap<LottoRanking, Int> {
```

Comment:
```suggestion
    fun lottoRanking(purchaseLottoNumbers: List<LottoNumbers>, winningLotto: WinningLotto): Map<LottoRanking, Int> {
```
외부에서는 가급적이면 Mutable한 타입을 받지 못하도록 하는 것이 좋습니다.
Map 타입으로 받도록 하더라도, Mutable로 변경해서 값을 변경하게 되면 원본이 훼손되기 때문입니다.
수업시간에 배운 "방어적 복사"를 활용해보면 어떨까요?

Reply:
수정하였는데, 방어적 복사를 활용하는 방법은 정확히 모르겠습니다.